File: updated_code/code/python/1752-check-if-array-is-sorted-and-rotated.py
Python
from typing import List

class Solution:
    """
    Given an array `nums`, return `true` if the array was originally sorted in
    non-decreasing order, then rotated **some** number of positions (including
    zero). Otherwise, return `false`.

    There may be **duplicates** in the original array.

    **Note:** An array `A` rotated by `x` positions results in an array `B` of the
    same length such that `A[i] == B[(i+x) % A.length]`, where `%` is the modulo
    operation.
    """

    def check(self, nums: List[int]) -> bool:
        """
        Check if the array was originally sorted in non-decreasing order, then
        rotated **some** number of positions (including zero).
        """
        if len(nums) <= 1:
            return True

        # Check if the array is sorted
        for i in range(1, len(nums)):
            if nums[i-1] > nums[i]:
                return False

        # Check if the array has been rotated
        if nums[0] >= nums[-1]:
            return False

        return True

File: updated_code/code/java/1752-check-if-array-is-sorted-and-rotated.java
package problem1752;

import java.util.Arrays;

class Solution {
    public boolean check(int[] nums) {
        // Check if the array is not rotated and still sorted in non-decreasing order
        if (Arrays.stream(nums).distinct().count() == 1) {
            Arrays.sort(nums);
            return Arrays.equals(nums, nums.clone()); // Check if the array is sorted
        }

        // Check if the array was rotated some number of positions
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] > nums[i + 1]) {
                // Check if the array was rotated with the smallest element as the pivot
                if (i == 0 || nums[i] < nums[i - 1]) {
                    return true;
                } else {
                    return false;
                }
            }
        }
        return nums[0] <= nums[nums.length - 1];
    }
}